tests/view/RoomsTest.java

```diff
@@ -0,0 +1,175 @@
+package view;
+
+import app.RoomsUseCaseFactory;
+import app.SearchUseCaseFactory;
+import data_access.ElasticsearchDataAccessObject;
+import data_access.FirebaseMessageDataAccessObject;
+import data_access.FirebaseRoomsDataAccessObject;
+import data_access.FirebaseUserDataAccessObject;
+import entities.auth.DisplayUser;
+import entities.auth.User;
+import entities.rooms.Message;
+import entities.rooms.Room;
+import interface_adapter.ViewManagerModel;
+import interface_adapter.rooms.RoomsState;
+import interface_adapter.rooms.RoomsViewModel;
+import interface_adapter.search.SearchController;
+import interface_adapter.search.SearchViewModel;
+import interface_adapter.searched.SearchedViewModel;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.swing.*;
+import okhttp3.OkHttpClient;
+import org.junit.Assert;
+import org.junit.Test;
+import use_case.search.SearchDataAccessInterface;
+
+public class RoomsTest extends ButtonTest {
+
+  @Test
+  public void testClickingRefresh() {
+    RoomsViewModel roomsViewModel = new RoomsViewModel();
+    OkHttpClient client = new OkHttpClient();
+    RoomsView roomsView =
+        RoomsUseCaseFactory.create(
+            new FirebaseRoomsDataAccessObject(client),
+            new FirebaseMessageDataAccessObject(client),
+            new FirebaseUserDataAccessObject(client),
+            roomsViewModel,
+            null,
+            null);
+
+    roomsViewModel.setState(this.buildTestState());
+
+    JButton refresh = roomsView.getRefreshButton();
+    refresh.doClick();
+
+    // User can load blank rooms with request to Firebase
+    Assert.assertTrue(roomsViewModel.getState().getError() == null);
```

Comment:
Similarly, I believe you can do assertNull here

Reply:
ok changing this